src/assets/js/map.js

```diff
@@ -86,49 +161,102 @@ class Map {
+      geometry.forEach(pullOutCoordinates)
+    } else {
+      const [x, y] = geometry
+
+      // if x or y isn't a valid number log an error and continue
+      if (isNaN(x) || isNaN(y)) {
+        console.error('Invalid coordinates', x, y)
+        return
       }
+
+      minX = Math.min(minX, x)
+      minY = Math.min(minY, y)
+      maxX = Math.max(maxX, x)
+      maxY = Math.max(maxY, y)
     }
+  }
+
+  pullOutCoordinates(geometries)
+
+  // Return the bounding box
+  return [[minX, minY], [maxX, maxY]]
+}
+
+const generatePaginatedGeoJsonLinks = async (geoJsonUrl) => {
+  const geoJsonLinks = [geoJsonUrl]
+  const initialResponse = await fetch(geoJsonUrl)
+  const initialData = await initialResponse.json()
+
+  // return if no pagination is needed
+  if (!initialData.links || !initialData.links.last) {
+    return geoJsonLinks
+  }
 
-    pullOutCoordinates(geometries)
+  const lastLink = new URL(initialData.links.last)
+  const limit = parseInt(lastLink.searchParams.get('limit'))
+  const lastOffset = parseInt(lastLink.searchParams.get('offset'))
 
-    // Return the bounding box
-    return [[minX, minY], [maxX, maxY]]
+  if (!limit || !lastOffset) {
+    console.error('Invalid pagination links', lastLink)
+    return geoJsonLinks
   }
+
+  // create a loop to generate the links
+  for (let offset = limit; offset < lastOffset; offset += limit) {
+    const newLink = new URL(geoJsonUrl)
+    newLink.searchParams.set('offset', offset)
+
+    geoJsonLinks.push(newLink.toString())
+  }
+
+  return geoJsonLinks
+}
+
+const generateBoundingBox = async (boundaryGeoJsonUrl) => {
+  const res = await fetch(boundaryGeoJsonUrl)
+  const boundaryGeoJson = await res.json()
+
+  return calculateBoundingBoxFromGeometries(boundaryGeoJson.features[0].geometry.coordinates)
 }
 
-const createMapFromServerContext = () => {
-  const { containerId, geometries, mapType } = window.serverContext
+const createMapFromServerContext = async () => {
+  const { containerId, geometries, mapType, geoJsonUrl, boundaryGeoJsonUrl } = window.serverContext
+  const options = {
+    containerId,
+    data: geometries,
+    boundaryGeoJsonUrl,
+    interactive: mapType !== 'static',
+    wktFormat: geoJsonUrl === undefined
+  }
+
+  // if the geoJsonUrl is provided, generate the paginated GeoJSON links
+  if (geoJsonUrl) {
+    options.data = await generatePaginatedGeoJsonLinks(geoJsonUrl)
+  }
+
+  if (options.boundaryGeoJsonUrl) {
+    options.boundingBox = await generateBoundingBox(boundaryGeoJsonUrl)
+  }
 
   // if any of the required properties are missing, return null
-  if (!containerId || !geometries) {
+  if (!options.containerId || !options.data) {
     console.log('Missing required properties (containerId, geometries) on window.serverContext', window.serverContext)
     return null
   }
 
-  return new Map(containerId, geometries, mapType !== 'static')
+  return new Map(options)
 }
 
 const newMap = createMapFromServerContext()
```

Comment:
Missing error handling: there's a bunch of promises being created via that function and no checks if any of them were rejected. We probably can get away with pretty coarse error handling here (e.g. one `.catch(...)` around the `createMapFromServerContext()` - not sure if there were discussions on how to let the user know we can't show the map because something went wrong).

Reply:
Done

Reply:
Feeding back to users is something we need to discuss with designers in a future iteration